app/models/physical_network_port.rb

```diff
@@ -1,4 +1,18 @@
 class PhysicalNetworkPort < ApplicationRecord
   belongs_to :guest_device
   belongs_to :physical_switch
+
+  #
+  # Get the peer port connected.
+  #   A connected port has the mac address of the pair in its
+  #   peer_mac_address field.
+  #
+  # @return [PhysicalNetworkPort] - the connected port
+  #
+  def connected_port
+    port = PhysicalNetworkPort.find_by('peer_mac_address IS NOT NULL AND peer_mac_address = ?', mac_address)
+    port = PhysicalNetworkPort.find_by('mac_address IS NOT NULL AND mac_address = ?', peer_mac_address) if port.blank?
```

Comment:
Hm not a fan of doing the queries every time we call `#connected_port`.  Is this something we can do during refresh?  E.g. could we set a peer_port_id?

Reply:
Hmm, good point, I can see the problem. I will do it in the refresh cycle. Thanks